fix Form getValues() invalid type

Bleksak opened the pull request:
This PR fixes an issue where if you want to map values into a DTO, you need to manually assert the type with assert instanceof.

This PR allows you to do `$values = $form->getValues(MyFormDto::class)` and the $values type will be properly typed.

Bleksak committed:
fix getValues() invalid type

ondrejmirtes commented:
Please add a type inference test about this. Here's an example https://github.com/phpstan/phpstan-nette/blob/2.0.x/tests/Type/Nette/ComponentModelArrayAccessDynamicReturnTypeExtensionTest.php and docs https://phpstan.org/developing-extensions/testing#type-inference.

Bleksak commented:
@ondrejmirtes Hi, thank you for your quick response. I added the test as required. I also quickly tested if getValues(true), getValues(false) still work and they both throw an exception at runtime, so I removed it from the original code.

ondrejmirtes commented:
Thank you for cleaning up that test! It was really old and now we know better than to mock the whole universe 😊
